source/libtomo/misc/median_filt3d.c

```diff
@@ -98,7 +100,8 @@ medfilt3D_float(float* Input, float* Output, int radius, int sizefilter_total,
                 k1 = k + k_m;
                 if((k1 < 0) || (k1 >= dimZ))
                     k1 = k;
-                ValVec[counter] = Input[(dimX * dimY) * k1 + j1 * dimX + i1];
+                index1 = (size_t)(dimX * dimY * k1 ) + (size_t)(j1 * dimX + i1);
```

Comment:
What's the pro/con of using `size_t` for all indexing and size variables instead of a mix of long and size_t with casting? Do some of these indexing variable need to represent negative integers?

My concern with mixing types and casting on lines like this is that it might still overflow before the cast to size_t? Because the order of operations is to perform the computation inside the parenthesis before casting.

Maybe it should be something like this? Where everything that is not cast is already `size_t`? This way the multiplications are computed using types that are already `size_t`.
```suggestion
                index1 = dimX * dimY * (size_t)k1 + (size_t)j1 * dimX + (size_t)i1;
```

Reply:
thanks, makes total sense. I guess `(size_t)(dimX * dimY * k1 )` could overflow before getting converted to `size_t`. will try to fix...